portal/server/controllers/dashboard/filters/generate-template-filters.js

```diff
@@ -12,17 +12,7 @@ const { SUBMISSION_TYPE } = require('../../../constants');
  * @returns { text: 'GEF deals', value: 'GEF', checked: true }
  */
 const generateFilterObject = (field, text, value, submittedFilters) => {
-  let checked = false;
-
-  const hasSubmittedFilters = Object.keys(submittedFilters).length;
-
-  if (hasSubmittedFilters) {
-    const filterHasBeenSubmitted = submittedFilters[field] && String(submittedFilters[field]).includes(String(value));
-
-    if (filterHasBeenSubmitted) {
-      checked = true;
-    }
-  }
+  const checked = !!(Object.keys(submittedFilters).length && submittedFilters[field] && String(submittedFilters[field]).includes(String(value)));
```

Comment:
`isChecked` please

Reply:
Also could we consider making this a little clearer?
```
const isFieldInSubmittedFilters = Object.keys(submittedFilters).length &&!!submittedFilters[field]; // does this have any issue with falsey things?
const isValueInField = isFieldInSubmittedFilters && String(submittedFilters[field]).includes(String(value));

const isChecked = isFieldInSubmittedFilters && isValueInField;
```
or something like this? Let me know what you think -- I'm finding this review a tad hard to dive into as I've never seen this code before

Reply:
on the `isChecked` point, there is some precedence to keep this the same since it will be passed straight into a nunjucks filter

Reply:
But i think id prefer this to be more readable than more compact, given the weirdness in this area

Reply:
Happy with whatever works for this area of the code